File: investigation-admin/src/api/statistics.ts
import { apiRequest } from "./client"

export type DailyStat = {
  stat_date: string
  tasks_created: number
  tasks_completed: number
  tasks_overdue: number
  pomodoro_count: number
  focus_minutes: number
  habits_completed: number
  active_tasks: number
  generated_at: string
}

export function fetchDailyStats(params?: Record<string, any>): Promise<DailyStat[]> {
  return apiRequest<DailyStat[]>("/statistics/daily", { params })
}

File: investigation-backend/src/schemas/statistics.py
from datetime import date, datetime

from pydantic import BaseModel


class DailyStat(BaseModel):
    stat_date: date
    tasks_created: int
    tasks_completed: int
    tasks_overdue: int
    pomodoro_count: int
    focus_minutes: int
    countup_count: int
    countup_minutes: int
    habits_completed: int
    active_tasks: int
    generated_at: datetime
